thunder/core/transform_common.py

```diff
@@ -33,6 +33,40 @@ def _remove_noop_subsymbols(bsym: BoundSymbol) -> None:
     bsym.subsymbols = nsbsyms
 
 
+def _inplace_copy_sanity_check(extrace: Trace):
+    """Make sure that the copy_to argument of prims.copy_ is not used as input for any of its subsequent operators, except for the Return and Del operators."""
```

Comment:
Please expand this docstring to include more details about the necessity of this check, why it's important to have this check, the types of neural network patterns that are permissible, the patterns this check aims to block, and the issues associated with those patterns.